### DISCUSSION TITLE
[Question] Multi-task RL with tasks having different observation terms

### DISCUSSION THREAD

**Discussion by meenalparakh** [isaac-sim/IsaacLab#811] [Q&A]
Hello! Is it possible to set up multi-task RL training (where the tasks have different numbers of objects and corresponding observation terms) using Isaac-lab and get any insights on how to approach it? For example, I want to train a single PPO policy on five such tasks. How can I collect data across multiple tasks efficiently? 
- Will it be very costly to spawn, say, 100 environments for a task and change the task after every few episodes? 
- What about spawning environments for all tasks side-by-side, in which case there is no need to create scenes from scratch, but then how can I initialize it (e.g., 100 envs for task#1, 100 envs for task#2 ... in the same world) and obtain the different sets of observation and actions associated with them? 
- Any other ways or resources to help with this?
Thank you!

**Comment by XintongHe** [comment]
Hi, did you solve your problems? I have the same problem. Thank you
